### ISSUE TITLE
neb_fesc + neb_fdust have no joint constraint: half the default prior gives identically zero nebular emission

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2436]
## Summary

`neb_fesc` and `neb_fdust` are two fractions of the **same** ionizing-photon budget, so
physically `f_esc + f_dust < 1`. They are declared as **independent** `Uniform(0, 1)`
priors, and nothing — not the declaration, not `SEDModel.build`, not any warning at fit
time — enforces or flags the joint constraint.

`lyc_dust_escape_factor` handles an out-of-range sum by clipping
(`src/tengri/components/nebular/_recombination_coeffs.py:121`):

```python
f_total = jnp.clip(f_esc + f_dust, 0.0, 1.0)
```

so for **any** `f_esc + f_dust >= 1` the numerator `1 - f_total` is exactly `0.0`, and all
nebular emission — lines *and* continuum — is multiplied by zero, silently.

Over the default independent `Uniform(0, 1)` priors that is **50.1 % of the joint prior
volume** (200k draws; 51.7 % has `k < 0.01`). The median `k` over the prior is exactly `0`.

## Why this is worse than a wasted half-prior

Inside that half, six *other* free parameters have **exactly zero gradient**, because
everything they shape has been multiplied by zero. Measured on a minimal
stellar + Cue model, no dust, `approx=None`, float64, at the prior medians
(`neb_fesc = neb_fdust = 0.5`, so `k = 0`):

```
d(sum photometry)/d(param):
   neb_dig_delta_logU        0.000000e+00   <-- ZERO GRADIENT
   neb_dig_frac              0.000000e+00   <-- ZERO GRADIENT
   neb_eline_sigma_kms       0.000000e+00   <-- ZERO GRADIENT
   neb_fdust                -1.699481e-31
   neb_fesc                 -1.699481e-31
   neb_fesc_lya              0.000000e+00   <-- ZERO GRADIENT
   neb_logU                  0.000000e+00   <-- ZERO GRADIENT
   neb_logZ_gas              0.000000e+00   <-- ZERO GRADIENT
```

So the posterior carries a **6-dimensional, exactly flat plateau over half the prior
volume**. For NUTS that is not a soft degeneracy to be absorbed by mass-matrix
adaptation: the gradient is identically zero and there is nothing to follow.

The Cue forward still runs in that region (`predict_nebular_sed` 2 calls,
`_forward_lines` 4, `_forward_continuum` 2 per evaluation) — the work is done and then
multiplied by zero, so this also costs the full Cue NN forward for a guaranteed-zero result.

## Reproduction

```python
import jax, jax.numpy as jnp, numpy as np, tengri
jax.config.update("jax_enable_x64", True)
from tengri import DEFAULT, FREE, Fixed, Observation, Photometry, SEDModel
from tengri.components.nebular._recombination_coeffs import lyc_dust_escape_factor

print(float(lyc_dust_escape_factor(0.5, 0.5)))   # 0.0        <- prior medians
print(float(lyc_dust_escape_factor(0.1, 0.1)))   # 0.71468
print(float(lyc_dust_escape_factor(0.0, 0.0)))   # 1.0

ssp = tengri.load_ssp("fsps_mist_c3k_a_chabrier")
m = SEDModel.build(
    ssp_data=ssp, observation=Observation(photometry=Photometry.from_names(["sdss_g"])),
    sfh={"type": "delayed", "all_params": Fixed(DEFAULT),
         "log_total_mass": 10.0, "tau_gyr": 1.0, "age_gyr": 5.0},
    dust_attenuation={"type": "none"}, neb={"type": "cue", "all_params": FREE},
    redshift=Fixed(1.0), approx=None)
free = sorted(m.spec.free_params)
base = {n: float(m.spec.get_distribution(n).unstandardize(0.0)) for n in free}
names = [n for n in free if n.startswith("neb")]
g = jax.grad(lambda v: jnp.sum(m.predict_photometry(
        {**base, **dict(zip(names, v))})))(jnp.array([base[n] for n in names]))
for n, gv in zip(names, np.asarray(g)):
    print(f"{n:<24}{gv: .6e}")
```

Nebular share of the `sdss_g` flux in the same model, varying only the pair:

```
  fesc=0.500 fdust=0.500 -> nebular is  0.0000% of sdss_g flux
  fesc=0.100 fdust=0.100 -> nebular is  6.1209%
  fesc=0.000 fdust=0.000 -> nebular is  8.3622%
```

## Not the FeaturePrecomp mechanism

This is **not** #1281 / #1748 (the per-Q_H grid serving photometry by zeroing
`sed_nebular`). Reproduced here with `approx=None`, `grid_table is None`,
`use_grid == False`, with and without dust attenuation and dust emission — identical
result in all four shapes. The two mechanisms share the symptom "nebular emission is
zero" and have nothing else in common.

## How it surfaced

A one-at-a-time sensitivity sweep over a 44-parameter joint mock (move each parameter
from its prior 5 % to 95 % quantile with the rest at truth, score
`sum[(F95-F05)/sigma]^2` over 15 detected bands + 1500 spectral pixels) returned
**exactly 0.000** for six nebular parameters. An exact zero in float64 is bit-identical
output, which is a mechanism, not a weak constraint — that is what led here.

That mock's truth was `neb_fesc = 0.7079`, `neb_fdust = 0.4624`, **sum = 1.1703**, so
`k = 0` and the mock built to demonstrate Cue nebular emission contained none. Both
values were drawn independently from their declared priors, and nothing warned.

## Docstring drift in the same function

The docstring states the sum "is clamped to ``1 - 1e-8`` to preserve numerical
stability" (`_recombination_coeffs.py:90`), but the code clamps to `1.0` (line 121).
With `1 - 1e-8` the factor would be ~`6.3e-9`; with `1.0` it is exactly `0`. The
"Gradient-safe" note (line 86) is about the *denominator*, which is indeed bounded — it
is the *numerator* that collapses. Whichever behaviour is intended, the two disagree.

## Possible directions

1. Reparameterize onto the simplex the physics actually lives on: one parameter for the
   total lost fraction `f_lost = f_esc + f_dust` in `[0, 1)` and one split fraction. The
   standardized transform then makes the constraint structural and the plateau
   unreachable, which fits the "prior lives in the forward model" design.
2. Failing that, a build-time warning when both are free with priors whose sum can reach
   1, naming the fraction of prior volume that yields `k = 0`.
3. At minimum, do not return exactly `0` silently: a fit-time warning when a MAP or a
   chain sits in the `k = 0` region would have caught the mock above immediately.

Happy to take direction on which of these is wanted before opening a PR.

Found at `aad3d8e00`.
